lib/junit.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module AutoHCK
+  class JUnit
+    def initialize(project)
+      @project = project
+      @logger = project.logger
+
+      template = Erubi::Engine.new(File.read('lib/templates/junit.xml.erb'), escape: true)
+      source = <<-RUBY
+        def build_xml
+          #{template.src}
+        end
+      RUBY
+
+      JUnit.module_eval(source)
```

Comment:
This should be moved out of the `initialize` method as it initializes the class instead of an instance.